lib/index.js

```diff
@@ -68,58 +65,30 @@ function isTruthy (x) {
  * @param {String} [options.textCharset] - charset for text email body
  * @param {String} [options.subjectCharset] - charset for email subject
  * @param {Function} callback - continuation function called after
- * the email has been sent. */
-
+ * the email has been sent.
+ * @returns {undefined}
+ */
 function sendMany (options, callback) {
   /* eslint-disable max-len */
-  // see: http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SES.html#sendEmail-property
-
-  var toAddresses = options.toAddresses || [];
-  var ccAddresses = options.ccAddresses || [];
-  var bccAddresses = options.bccAddresses || [];
-  var defaultCharset = 'utf8';
-  var params;
-
-  toAddresses = toAddresses.filter(isTruthy);
-  ccAddresses = ccAddresses.filter(isTruthy);
-  bccAddresses = bccAddresses.filter(isTruthy);
-
-  params = {
-    Destination: {
-      /* required */
-      BccAddresses: bccAddresses,
-      CcAddresses: ccAddresses,
-      ToAddresses: toAddresses
-    },
-    Message: {
-      /* required */
-      Body: {
-        /* required */
-        Html: {
-          Data: compile_template(options.templateName, 'html')(options.context),
-          /* required */
-          Charset: options.htmlCharset || defaultCharset
-        },
-        Text: {
-          Data: compile_template(options.templateName, 'txt')(options.context),
-          /* required */
-          Charset: options.textCharset || defaultCharset
-        }
-      },
-      Subject: {
-        /* required */
-        Data: options.subject,
-        /* required */
-        Charset: options.subjectCharset || defaultCharset
-      }
-    },
-    Source: options.senderEmailAddress || process.env.SENDER_EMAIL_ADDRESS, /* required */
-    ReplyToAddresses: [
-      options.replyToAddress || process.env.SENDER_EMAIL_ADDRESS
-    ]
-  };
 
-  ses.sendEmail(params, callback);
+  // Deep clones options, ensures no mutation of input (deep necessary because source may contain arrays)
+  var clean_options = JSON.parse(JSON.stringify(options));
```

Comment:
@zemccartney could you just explain why a deep clone was needed here? As far as I'm aware we weren't mutating the options passed in before. At which point did this become necessary? - I just think it would be great if it was possible to avoid this.

(I quite liked how easy these lines were to read before: https://github.com/dwyl/sendemail/blob/master/lib/index.js#L77-L85)

Reply:
@samhstn hey, thanks calling this out. definitely a bit of a kludge. my bad on that.

I used a deep clone here because I needed to end up with an object containing all the values of the `options` object that I could then pass as the first parameter to the `send` call. Since we're running those conditional checks and filters on the properties of that object and need to store those results, I _think_ I thought that meant we'd be mutating the input `options` (sorry, it's been long enough since I wrote this that I've since forgotten / am a bit lost myself on what I was thinking :) ). 

I tried to clean this up a good bit in the work I just pushed up. Still creating a new object, but not cloning our options object. Hopefully readable, little less misdirection here. How does that look?